eslint.config.js

```diff
@@ -0,0 +1,69 @@
+import eslintjs from '@eslint/js';
+import eslintConfigPrettier from 'eslint-config-prettier';
+import { configs as tseslintConfigs } from 'typescript-eslint';
+import globals from 'globals';
+
+const { configs: eslintConfigs } = eslintjs;
+
+const sourceFiles = ['bench/**/*.js', 'scripts/**/*.ts', 'packages/*/lib/**/*.ts'];
+const testFiles = ['test/**/*.{ts,js}'];
+const ignoreFiles = [
+    'test/data/html5lib-tests',
+    'test/data/html5lib-tests-fork',
+    'packages/*/dist/',
+    'test/dist/',
+    'docs/build/',
+];
+const allFiles = [...sourceFiles, ...testFiles];
+
+export default [
+    {
+        files: allFiles,
+    },
+    {
+        ignores: ignoreFiles,
+    },
+    {
+        languageOptions: {
+            globals: {
+                ...globals.nodeBuiltin,
+                ...globals.es2019,
+            },
+        },
+    },
+    eslintConfigs.recommended,
+    ...tseslintConfigs.recommended,
+    {
+        rules: {
+            'no-console': 'error',
+            curly: ['error', 'all'],
+            'prefer-arrow-callback': 'error',
+            'one-var': ['error', 'never'],
+            'no-var': 'error',
+            'prefer-const': 'error',
+            'object-shorthand': 'error',
+            'prefer-destructuring': [
+                'error',
+                {
+                    object: true,
+                    array: false,
+                },
+            ],
+            'prefer-template': 'error',
+            'arrow-body-style': ['error', 'as-needed'],
+        },
+    },
+    {
+        files: ['**/*.ts'],
+        rules: {
+            '@typescript-eslint/no-unsafe-declaration-merging': 'off',
+            '@typescript-eslint/no-non-null-assertion': 'warn',
+            '@typescript-eslint/no-explicit-any': 'warn',
+            '@typescript-eslint/explicit-function-return-type': 'error',
+            '@typescript-eslint/consistent-type-imports': 'error',
+
+            '@typescript-eslint/no-unused-vars': ['error', { argsIgnorePattern: '^_' }],
+        },
+    },
+    eslintConfigPrettier,
+];
```

Comment:
This looks like it is removing `plugin:unicorn/recommended`? Why?

Reply:
It is

Much of it is covered by the formatter or the recommended rule sets

It would be good to know what exactly we wanted it to catch. It's a very big plugin for not much benefit by the looks of it

Reply:
Re the first paragraph, much of ESlint can also be done by hand, yet we still use ESlint.

Re the second, that is very subjective, and I strongly disagree.
Removing lint rules is always silent. The project was formatted. So of course you don‘t notice that lint rules are useful.

Re the last paragraph, I even stronger disagree. Lint rules are very useful in my experience.

Reply:
The last paragraph was very far from suggesting lint rules are not useful. We pulled a very large package in without really knowing which rules we cared about other than the ones we manually enabled (i.e I don't think any of us ever knew what was in the recommended config) 

Much of it is stylistic and should be owned by the formatter (and already is). But if there are other rules in there we think are useful, I can reintroduce it

The recommended configs have grown since and cover all the more important stuff.

Reply:
@fb55 has intentionally pulled these rules in. I agree with these rules. Many rules are not things prettier does: https://github.com/sindresorhus/eslint-plugin-unicorn#rules. This change has nothing to do with the PR title.

Reply:
The PR removes many of our customisations, so it certainly was relevant

I don't care enough either way so I will reintroduce it some day when I get time and move past this

Once a more focused eslint package exists, maybe we can drop it for good. Overly generic, unfocused plugins are not great

Reply:
This PR is titled “upgrade to eslint”, not “remove things I find irrelevant”.
I do care.
I don’t think you are correct in calling things you dislike unfocussed or overly generic.
I am taken aback that you want to push things through that don’t have consensus.

@fb55 Please chime in.

How is this project governed?

Reply:
Just because unicorn is broad does not mean it doesn't have value 😄

I am generally a fan of [Isaac Schlueter's Principle of Most Restrictive Production](https://blog.izs.me/2022/04/principle-of-most-restrictive-production) and enjoy when lint tools push a codebase in that direction. Unicorn has a number of rules that help with that, eg. for regex, or switch statements.

Most of the interesting unicorn rules have been ported to Biome, so that might be an alternative we could migrate to.

@wooorm Let's not dig in to each other too much, this is how maintainers burn out 😅

Reply:
The original post specifies that I simplified the customisations we have. It was part of that. I would have removed some of the custom rule options too but it turns out they're not in the recommended rule set anymore. 

It was part of the described change regardless of the title. Apologies for not having a longer title

Too much energy has been lost discussing this as I don't care enough about removing it to take part in a debate

I will very simply reintroduce it 👍

Reply:
@fb55 I would still appreciate info on how this project is governed. Consensus-seeking? Vote? Consensus with a vote as backup? Dictator?
Was there something agreed with Ivan?
As you mention maintainer burnout, and you once asked me to help, and there is no `governance.md`, I’d like to hear your thoughts.

Reply:
No agreement with Ivan. The three of us are responsible for the project, and should try to move it forward in good faith. There has been little activity in the repo for the past year, and setting up a proper governance structure does not seem like an immediate need.

Super happy to see @43081j push things forward, and I do believe that we should support these efforts instead of fighting for small stuff.